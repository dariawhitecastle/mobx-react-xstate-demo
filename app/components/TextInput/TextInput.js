/* eslint-disable react/prop-types */
/**
 *
 * TextInput
 *
 */

import React from 'react'
import { TextInput } from 'grommet'
import { doc } from './TextInput.doc'
import helpers from '../../utils/helpers'

function createTextInput({
  dropAlign,
  dropHeight,
  dropProps,
  dropTarget,
  focusIndicator,
  id,
  messages,
  name,
  onChange,
  onSelect,
  placeholder,
  plain,
  size,
  value,
  suggestions,
  ...rest
}) {
  return (
    <TextInput
      dropAlign={dropAlign}
      dropHeight={dropHeight}
      dropProps={dropProps}
      dropTarget={dropTarget}
      focusIndicator={focusIndicator}
      id={id}
      messages={messages}
      name={name}
      onChange={onChange}
      onSelect={onSelect}
      placeholder={placeholder}
      plain={plain}
      size={size}
      value={value}
      suggestions={suggestions}
      {...rest}
    />
  )
}

export default helpers.createWithDoc({
  docFunction: doc,
  component: createTextInput,
})
