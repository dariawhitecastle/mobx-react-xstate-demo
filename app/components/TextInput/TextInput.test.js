/**
 *
 * Tests for TextInput
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { fireEvent, render } from '@testing-library/react'
// import 'jest-dom/extend-expect'; // add some helpful assertions

import TextInput from './TextInput'

const renderComponent = (props = {}) => render(<TextInput {...props} value="" />)
const setup = () => {
  const utils = render(<TextInput placeholder="text-input" />)
  const input = utils.getByPlaceholderText('text-input')
  return {
    input,
    ...utils,
  }
}

test('it renders and matches snapshot', () => {
  const { container } = renderComponent()
  expect(container).toMatchSnapshot()
})

test('It should update value on input', () => {
  const { input } = setup()
  expect(input.value).toBe('') // empty before
  fireEvent.change(input, { target: { value: 'Good Day' } })
  expect(input.value).toBe('Good Day')
})
