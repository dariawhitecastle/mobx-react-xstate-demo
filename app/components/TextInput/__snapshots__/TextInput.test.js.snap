// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches snapshot 1`] = `
<div>
  <div
    class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
  >
    <input
      autocomplete="off"
      class="StyledTextInput-sc-1x30a0s-0 kvYrcH"
      value=""
    />
  </div>
</div>
`;
